pkg/lease/service.go

```diff
@@ -129,6 +139,81 @@ func (a *Service) List(query *Lease) (*Leases, error) {
 	return leases, nil
 }
 
+// Create creates a new lease using the data provided. Returns the lease record
+func (a *Service) Create(data *Lease, principalSpentAmount float64) (*Lease, error) {
```

Comment:
Passing in `principalSpentAmount` into the `Create` method seems awkward here. It seems like it should only take a Lease, like the account create only takes an account create, and that something (like a controller) should have stopped it before getting this far.

Reply:
Yes, I wanted to keep the business validation check in service layer. And reaching out to usagesvc in controller.

Reply:
Are we doing the business validation checks anywhere else in the service layer? It seems like that's something the controllers have been doing.